src/main/java/com/aws/greengrass/clientdevices/auth/ClientDevicesAuthService.java

```diff
@@ -198,21 +203,47 @@ protected void install() throws InterruptedException {
                 }
             }
 
+            Topics deviceGroupTopics = this.config.lookupTopics(CONFIGURATION_CONFIG_KEY, DEVICE_GROUPS_TOPICS);
+            Topics certAuthorityTopic = this.config.lookupTopics(CONFIGURATION_CONFIG_KEY, CERTIFICATE_AUTHORITY_TOPIC);
+            String caKeyPath = Coerce.toString(certAuthorityTopic.findOrDefault("", CA_PRIVATE_KEY_URI));
+            String caCertPath = Coerce.toString(certAuthorityTopic.findOrDefault("", CA_CERTIFICATE_URI));
+            Topic caType = certAuthorityTopic.lookup(CA_TYPE_TOPIC);
+
+
             if (whatHappened == WhatHappened.initialized || node == null) {
                 updateDeviceGroups(whatHappened, deviceGroupTopics);
-                updateCAType(caTypeTopic);
+                updateCertificateAuthority();
             } else if (node.childOf(DEVICE_GROUPS_TOPICS)) {
                 updateDeviceGroups(whatHappened, deviceGroupTopics);
-            } else if (node.childOf(CA_TYPE_TOPIC)) {
-                if (caTypeTopic.getOnce() == null) {
+            } else if (node.childOf(CERTIFICATE_AUTHORITY_TOPIC)) {
+                // If the CA type is null because of the change to its configuration to null, do not update the CA.
+
+                // If the CA type is null and if either of the customer provided CA key and CA certificate (or both)
+                // are empty, do not update the CA.
+                if (caType.getOnce() == null
+                        && (node.childOf(CA_TYPE_TOPIC) || Utils.isEmpty(caKeyPath) || Utils.isEmpty(caCertPath))) {
                     return;
                 }
-                updateCAType(caTypeTopic);
+                updateCertificateAuthority();
             }
         });
 
     }
 
+    private void updateCertificateAuthority() {
+        // Upload the generated or provided CA certificates to the GG cloud and update config
+        try {
+            List<String> caCerts = certificateManager.getCACertificates();
+            if (caCerts.isEmpty()) {
+                serviceErrored(new Exception("Unable to get the CA certificates"));
+            }
+            uploadCoreDeviceCAs(caCerts);
+            updateCACertificateConfig(caCerts);
+        } catch (KeyStoreException | IOException | CertificateEncodingException e) {
+            serviceErrored(e);
```

Comment:
It's not clear from this code what exceptions you're catching and what you're not. This should be refactored to improve clarity. For example, remove `uploadCoreDeviceCAs` from the try block since you're not catching the exception it throws. 
Clean up the logic such that we retry but continue. The customer may not be relying on cloud discovery at all, so IMO we should continue running.